lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -248,9 +252,10 @@ def add_creator_info(post, options)
       end
 
       def add_amount(post, money, options = {})
+        currency_ogone = 'EUR' if ogone_direct?
         post['amountOfMoney'] = {
           'amount' => amount(money),
-          'currencyCode' => options[:currency] || currency(money)
+          'currencyCode' => currency_ogone || options[:currency] || currency(money)
```

Comment:
⚠️  Warning:

The problem with this approach is that you are forcing all Ogone direct users to use EUR in the future and with that limiting the future possibility for direct handling USD or other currencies

Reply:
Hi @Heavyblade you're right, I did a small change in order to give the opportunity to get first a currency if was specified. thanks